PizzaBuilder/PizzaBuilder/Program.cs

```diff
@@ -0,0 +1,188 @@
+        public WheatPizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+            this._wheatpizza = new WheatPizza();
+        }
+        public void BuildDough()
+        {
+            this._wheatpizza.Add("Prepare the dough");
+        }
+        public void BuildSauce()
+        {
+            this._wheatpizza.Add(" ، Add the sauce");
+        }
+        public void BuildTopping()
+        {
+            this._wheatpizza.Add(" ، Add Topping .");
+        }
+        public WheatPizza GetWheatPizza()
+        {
+            WheatPizza result = this._wheatpizza;
+            this.Reset();
+            return result;
+        }
+    }
+
+    public class WhitePizza
+    {
+        private List<object> pizza = new List<object>();
+        public void Add(string part)
+        {
+            this.pizza.Add(part);
+        }
+        public string ReadyPizza()
+        {
+            string str = "";
+            for (int i = 0; i < this.pizza.Count; i++)
+            {
+                str += this.pizza[i];
+            }
+            return "White Pizza ingredients : " + str;
+        }
+
+    }
+
+    public class WheatPizza
+    {
+        private List<object> pizza = new List<object>();
+        public void Add(string part)
+        {
+            this.pizza.Add(part);
+        }
+        public string ReadyPizza()
+        {
+            string str = "";
+            for (int i = 0; i < this.pizza.Count; i++)
+            {
+                str += this.pizza[i];
+            }
+            return "Wheat Pizza ingredients : " + str;
+        }
+
+    }
+
+
+    public class Director
+    {
+        private IBuilder _whitepizza;
+        public IBuilder whitepizzaBuilder
+        {
+            set { this._whitepizza = value; }
+        }
+
+        public void BuildMin_white()
+        {
+            this._whitepizza.BuildDough();
+            this._whitepizza.BuildSauce();
+        }
+        public void BuildFull_white()
+        {
+            this._whitepizza.BuildDough();
+            this._whitepizza.BuildSauce();
+            this._whitepizza.BuildTopping();
+        }
+
+        private IBuilder _wheatpizza;
+        public IBuilder wheatpizzaBuilder
+        {
+            set { this._wheatpizza = value; }
+        }
+        public void BuildMin_wheat()
+        {
+            this._wheatpizza.BuildDough();
+            this._wheatpizza.BuildSauce();
+        }
+        public void BuildFull_wheat()
+        {
+            _wheatpizza.BuildDough();
+            this._wheatpizza.BuildSauce();
+            this._wheatpizza.BuildTopping();
+        }
+    }
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("-*-*- white Yummy pizza -*-*-");
+            //client
+            var director_1 = new Director();
+            var whitepizza_builder = new WhitePizzaBuilder();

```

Comment:
It would be better if you used polymorphism to define builders, without creating new builder variable for each one.